### PR TITLE
XOR-156 [FEAT] Add support for configuring appearance for read-only fields

### PR DESCRIPTION
**Pull request by NikitaJadhavXoriant**
**Product areas affected**

Widgets -> Form Widget -> Add field -> Readonly field

**What does this PR do?**

Implemented functionality to achieve read only form field behavior

**JIRA ticket**

[click here](https://weboo.atlassian.net/browse/XOR-156)

**Result**

[XOR-Test-156All.webm](https://user-images.githubusercontent.com/108272606/190090809-788e8fd3-2559-4569-b3e8-8b2d994ce7e1.webm)

**Checklist**

 None

**Testing instructions**

None

**Deployment instructions**

None

Author concerns

None

### CODE REVIEW

File: scss/components/forms.scss [hunk 1]
```diff
@@ -1295,7 +1389,33 @@
...
-    [_type="flStarRating"] .rating-input:checked~.rating-star .fa {
+    [_type="flStarRating"] .rating-input~.readonly.rating-star .fa {
+      &:before {
+        content: "\f005";
+        -webkit-text-stroke: 1px map-get($configuration, formReadOnlySelectedFillColor);
+      }
+
+      // Styles for tablet
+      @include above($tabletBreakpoint) {
+        content: "\f005";
```
Comment (tonytlwu) [inline scss/components/forms.scss:1400]: @NikitaJadhavXoriant There shouldn't be any property in theme SCSS files that do not use at least one variable from the theme.

If it's not using any variable like this line, it should go into the appropriate library/widget CSS file. In this case, it looks like it should go into the Form widget CSS files.

Please verify this for the entire PR.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Review by amolsuryawanshi47 — APPROVED** [review]

File: scss/components/forms.scss [hunk 2]
```diff
@@ -1391,26 +1391,21 @@
...
-        content: "\f005";
+        &:before {
         -webkit-text-stroke: 1px map-get($configuration, formReadOnlySelectedFillColorTablet);
+        }
       }
 
       // Styles for desktop
       @include above($desktopBreakpoint) {
-        content: "\f005";
+        &:before {
```
Comment (tonytlwu) [inline scss/components/forms.scss:1406]: Indentation doesn't meet coding standards. I implemented and merged https://github.com/Fliplet/fliplet-theme-default/pull/198 which adds ESLint and applies the changes from [the formatter](https://gist.github.com/tonytlwu/8decaec74de85f118ed2b8dcf71dfbba) (see `[scss]` settings).

You need to:

1. Merge the changes from `master` to this branch so it's all up to date
2. Check your VSCode settings
3. Verify your changes to make sure they're correct and follow coding standards
4. Test your changes

**Review by tonytlwu — CHANGES_REQUESTED** [review]

**Review by amolsuryawanshi47 — APPROVED** [review]
Changes Looks ok to me

File: scss/components/forms.scss (hunk 1 above)
Comment (NikitaJadhavXoriant) [reply scss/components/forms.scss]: Done the changes accordingly and added in PR

File: scss/components/forms.scss (hunk 2 above)
Comment (NikitaJadhavXoriant) [reply scss/components/forms.scss:1482]: Done Changes accordingly, taken latest, added VS Code settings.

**Review by tonytlwu — APPROVED** [review]

**Pull request merged**